pkg/sm/sm.go

```diff
@@ -259,6 +266,10 @@ func (smb *ServiceManagerBuilder) Build() *ServiceManager {
 		log.C(smb.ctx).Panic(err)
 	}
 
+	if err := smb.calculateIntegrity(); err != nil {
```

Comment:
In case we decide to update the integrity as part of the application: we might try to acquire a lock, before the update, so only one instance of SM updates the integrity

Reply:
Also, the first time integrity is calculated, other instances should not start, but wait for lock. Otherwise they will start to serve requests with empty integrity for some resources

Reply:
How can this happen? Say node1 starts before node2:
* node1 starts processing integrity
* node2 starts processing the same integrity
* node1 finishes and starts processing requests - here all resources are with set integrity
* node2 continues to update integrity - it might do unnecessary db calls to update the integrity which should be okay.

Reply:
You are right